courses/sync_external_courses/external_course_sync_api.py

```diff
@@ -678,18 +681,23 @@ def create_or_update_external_course_run(course, external_course):
     is_created = is_updated = False
 
     if not course_run:
-        course_run = CourseRun.objects.create(
-            external_course_run_id=external_course.course_run_code,
-            course=course,
-            title=external_course.course_title,
-            courseware_id=course_run_courseware_id,
-            run_tag=external_course.course_run_tag,
-            start_date=external_course.start_date,
-            end_date=external_course.end_date,
-            enrollment_end=external_course.enrollment_end,
-            live=True,
-        )
-        is_created = True
+        try:
+            course_run = CourseRun.objects.create(
+                external_course_run_id=external_course.course_run_code,
+                course=course,
+                title=external_course.course_title,
+                courseware_id=course_run_courseware_id,
+                run_tag=external_course.course_run_tag,
+                start_date=external_course.start_date,
+                end_date=external_course.end_date,
+                enrollment_end=external_course.enrollment_end,
+                live=True,
+            )
+            is_created = True
+        except ValidationError as e:
+            log.error(
+                f"Error creating course run for course: {course.readable_id}, course run code: {external_course.course_run_code}, error: {e}"  # noqa: G004
+            )
```

Comment:
Instead of trying to create a course run with bad data, can we validate the dates like we are validating the end date in update_external_course_runs? We won't need these extra changes and it would be much simpler. What are your thoughts?